File: Proyects/UniJava/Workshops/WS3/FixedSubwayCard.java
public class FixedSubwayCard extends SubwayCard{
	private int rides;

	public FixedSubwayCard(SubwayCentral sc, int rides){
		super(sc);
		this.rides = rides;
	}

	@Override
	protected boolean canRide(){
		return (this.rides > 0);
	}

	@Override
	public void ride(){
		if(!canRide()){
			System.err.println("Error, cant ride");
		}
		this.rides--;
	}
}

File: Proyects/UniJava/TP8/ex6/fixed_subway_card.rb
require_relative 'card'

class FixedSubwayCard < Card
  def initialize(subway_central, cant_rides)
    @subway_central = subway_central
    @cant_rides = cant_rides
  end

  def ride
    if(@cant_rides == 0)
      raise "No rides available"
    end
    @cant_rides -= 1
  end

  def to_s
    "Fixed Subway Card [#{@subway_central.to_s}] [Rides: #{@cant_rides}]"
  end
end
